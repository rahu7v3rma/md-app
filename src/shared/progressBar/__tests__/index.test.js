import React from 'react';
import { Animated, View } from 'react-native';
import { act, create } from 'react-test-renderer';

import ProgressBar from '@/shared/progressBar';

beforeEach(() => {
    jest.useFakeTimers();
});

afterEach(() => {
    jest.useRealTimers();
});

test('snapshot/should render without errors', async () => {
    let tree;
    await act(() => (tree = create(<ProgressBar />)));
    expect(tree.toJSON()).toMatchSnapshot();
});

test('should display the initial progress correctly', async () => {
    let tree;
    await act(() => (tree = create(<ProgressBar />)));
    const width = tree.root.findByType(Animated.View).props.style[3].width;
    expect(width._parent._value).toBe(0);
    expect(width._parent._animation._toValue).toBe(0);
});

test('should update the progress animation', async () => {
    let tree;
    await act(() => (tree = create(<ProgressBar progress={10} />)));
    let width = tree.root.findByType(Animated.View).props.style[3].width;
    expect(width._parent._value).toBe(0);
    expect(width._parent._animation._toValue).toBe(10);
    await act(() => jest.advanceTimersByTime(1000));
    expect(width._parent._value).toBe(10);
    expect(width._parent._animation).toBe(null);
    await act(() => tree.update(<ProgressBar progress={20} />));
    width = tree.root.findByType(Animated.View).props.style[3].width;
    expect(width._parent._value).toBe(10);
    expect(width._parent._animation._toValue).toBe(20);
    await act(() => jest.advanceTimersByTime(1000));
    expect(width._parent._value).toBe(20);
    expect(width._parent._animation).toBe(null);
});

test('should apply the style prop correctly', async () => {
    let tree;
    await act(
        () => (tree = create(<ProgressBar style={{ borderColor: 'blue' }} />))
    );
    const view = tree.root.findByType(View);
    expect(view.props.style[1]).toMatchObject({ borderColor: 'blue' });
});

test('should apply the barStyle prop correctly', async () => {
    let tree;
    await act(
        () =>
            (tree = create(<ProgressBar barStyle={{ borderColor: 'blue' }} />))
    );
    const view = tree.root.findByType(Animated.View);
    expect(view.props.style[2]).toMatchObject({ borderColor: 'blue' });
});
