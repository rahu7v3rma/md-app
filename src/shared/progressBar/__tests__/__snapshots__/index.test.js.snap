// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot/should render without errors 1`] = `
<View
  style={
    [
      {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 8,
        "height": 5.6,
        "justifyContent": "flex-start",
        "overflow": "hidden",
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#4DBA0B",
        "borderBottomColor": "#3A9A00",
        "borderBottomWidth": 2,
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": "0%",
      }
    }
  />
</View>
`;
